### PR TITLE
Rf/unsupported check

### PR DESCRIPTION
**Pull request by electrachong**
Dependent on https://github.com/scality/Arsenal/pull/278

Moving check for unsupported queries and headers from routes to api.
The routes are a shared module between s3 projects which may diverge in their support based on different parameters (request method, etc).

This also means we won't have to pass another param to routes (were passing unsupportedQueries)

### CODE REVIEW

**Comment by electrachong** [comment]
@ironman-machine 
SCALITY_ARSENAL_BRANCH=rf/unsupportedCheck

**Comment by electrachong** [comment]
end-to-end retried here: http://ci.ironmann.io/gh/scality/Integration/13690

File: constants.js [hunk 1]
```diff
@@ -87,6 +87,17 @@ const constants = {
         'restore': true,
         'torrent': true,
     },
+
+    // Headers supported by AWS that we do not currently support.
+    encryptionHeaders: [
```
Comment (LaurenSpiegel) [inline constants.js]: I would generalize to unsupportedHeaders.  I would do an object for both as well.  Or as discussed an array if we loop through this and then check the reqHeaders.

File: lib/api/api.js [hunk 2]
```diff
@@ -50,6 +51,18 @@ auth.setHandler(vault);
...
+
+        const validationRes =
+            validateQueryAndHeaders(request.method, request.query,
+                request.headers, log);
+        if (validationRes.error) {
+            log.debug('request query / header validation failed', {
+                error: validationRes.error,
+                method: 'api.callApiMethod',
+            });
+            return callback(validationRes.error);
```
Comment (nicolas2bert) [inline lib/api/api.js]: We might need: `proccess.nextTick` here but also for the errors after.
So you might want to add them in another commit.

File: constants.js (hunk 1 above)
Comment (electrachong) [inline constants.js]: updated

File: lib/api/api.js (hunk 2 above)
Comment (electrachong) [inline lib/api/api.js]: updated

**Comment by electrachong** [comment]
after changes
@ironman-machine
SCALITY_ARSENAL_BRANCH=rf/unsupportedCheck

**Comment by electrachong** [comment]
end-to-end passed

**Pull request merged**